subprojects/dependency-management/src/main/java/org/gradle/api/internal/artifacts/transform/AttributeMatchingVariantSelector.java

```diff
@@ -101,130 +96,78 @@ public ResolvedArtifactSet select(ResolvedVariantSet producer, Factory factory)
     private ResolvedArtifactSet doSelect(ResolvedVariantSet producer, Factory factory, AttributeMatchingExplanationBuilder explanationBuilder) {
         AttributeMatcher matcher = schema.withProducer(producer.getSchema());
         ImmutableAttributes componentRequested = attributesFactory.concat(requested, producer.getOverriddenAttributes());
-        final Set<ResolvedVariant> variants;
+        final List<ResolvedVariant> variants;
         if (selectFromAllVariants) {
-            variants = producer.getAllVariants();
+            variants = ImmutableList.copyOf(producer.getAllVariants());
```

Comment:
The reason for this change is to get a `List` from a `Set`. `ImmutableList.copyOf` is just an easy way to do that -- though `new ArrayList<>()` is another option. If we want to avoid this copy here, we would need to use a `List` further up

Reply:
Yeah, maybe this is something to try to cleanup another day.